include/deal.II/matrix_free/operators.h

```diff
@@ -1103,10 +1114,19 @@ namespace MatrixFreeOperators
 
 
   template <typename OperatorType>
+  template <typename VectorType>
   void
-  MGInterfaceOperator<OperatorType>::Tvmult (LinearAlgebra::distributed::Vector<typename MGInterfaceOperator<OperatorType>::value_type> &dst,
-                                             const LinearAlgebra::distributed::Vector<typename MGInterfaceOperator<OperatorType>::value_type> &src) const
+  MGInterfaceOperator<OperatorType>::Tvmult (VectorType &dst,
+                                             const VectorType &src) const
   {
+#ifdef DEAL_II_WITH_CXX11
+#ifndef DEAL_II_MSVC
```

Comment:
That seems like a great idea. To stay conforming with other macros, I would call it `DEAL_II_STATIC_ASSERT`.